batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -258,14 +322,68 @@ static void runFhirFetch(FhirEtlOptions options, FhirContext fhirContext) throws
 		p.run().waitUntilFinish();
 	}
 	
-	public static void main(String[] args) throws CannotProvideCoderException {
+	static void runFhirJdbcFetch(FhirEtlOptions options, FhirContext fhirContext) throws PropertyVetoException, IOException {
+		Pipeline pipeline = Pipeline.create(options);
+		JdbcFhirMode jdbcUtil = new JdbcFhirMode();
+		JdbcIO.DataSourceConfiguration jdbcConfig = jdbcUtil.getJdbcConfig(options.getJdbcDriverClass(),
+		    options.getJdbcUrl(), options.getDbUser(), options.getDbPassword(), options.getJdbcMaxPoolSize());
+		
+		int batchSize = options.getBatchSize();
+		int parallelRequests = options.getParallelRequests();
+		ParquetUtil parquetUtil = new ParquetUtil(options.getOutputParquetBase());
+		LinkedHashMap<String, String> reversHashMap = jdbcUtil.createFhirReverseMap(options.getSearchList(),
+		    options.getTableFhirMapPath());
+		// process each table-resource mappings
+		for (Map.Entry<String, String> entry : reversHashMap.entrySet()) {
+			String tableName = entry.getKey();
+			String resourceType = entry.getValue();
+			String baseBundleUrl = options.getServerUrl() + options.getServerFhirEndpoint() + "/" + resourceType;
+			Schema schema = parquetUtil.getResourceSchema(resourceType);
+			
+			PCollection<GenericRecord> genericRecords = pipeline
+			        // fetch maxId for each passed resource
```

Comment:
Suggestion: This and the following line comments are probably useful at their corresponding classes as javadoc as well.

Reply:
thanks, will add in as javadocs